security-privacy-questionnaire.md

```diff
@@ -0,0 +1,73 @@
+
+This feature exposes files and directories the user explicitly selects to share with web sites with those web sites. This feature doesn't expose any more information than is already exposed via `<input type=file>` and `<input type=file webkitdirectory>` today.
+
+However this feature does make it possible for browsers to extend the time web sites have access to this information. I.e. permission grants and handles received through this API could be persisted giving web sites access to the same files/directories on disk when a user later returns to the website (or when a service worker for the origin is processing an event). At least for the Chrome implementation, we're only planning on having these grants persist for installed PWAs. On the drive-by web, access will only be as long as the web site is open, requiring re-prompting in subsequent visits.
+
+### 2.2. Is this specification exposing the minimum amount of information necessary to power the feature?
+
+Yes, we're only exposing files and directories explicitly selected by the user. Of course, web sites could ask for access to a directory when all it needs is access to some files, but the same is already true today. At least as far as exposing information to web sites is concerned, this API doesn't expose any more information than existing APIs do today.
+
+### 2.3. How does this specification deal with personal information or personally-identifiable information or information derived thereof?
+
+No data is exposed without the user explicitly choosing what files or directories to expose to the web site.
+
+### 2.4. How does this specification deal with sensitive information?
+
+No data is exposed without the user explicitly choosing what files or directories to expose to the web site, so only sensitive data that the user explicitly decides to share via this API will be shared. Furthermore, this API is only exposed in secure contexts, and third-party iframes (i.e. iframes that are cross origin from the top-level frame) won't be able to show pickers or permission prompts and can only access data they were already granted access to from a top-level same origin frame.
+
+### 2.5. Does this specification introduce new state for an origin that persists across browsing sessions?
+
+Yes, this specification lets websites store handles they've gotten access to (via a file or directory picker) in IndexedDB. User agents could also persist the permission grants that go with these handles, but at least in the Chrome implementation, these permission grants will only be persistent for installed PWAs. The drive-by web will only have enough state to allow it to re-prompt for access, but the access itself won't be persistent.
+
+Furthermore, the user will be able to clear storage (storage is just IndexedDB) and/or revoke permissions to clear the state that was persisted, similarly to how other permissions work.
```

Comment:
(storage is file handles in IndexedDB)

Would it be worth writing down the principle that we consider the files written by this API are owned by the user, not by the application, and making it explicit that we don’t recommend deleting the files written by this API?